File: CSharp/Leaderboard.Tests/LeaderboardTest.cs
using System.Collections.Generic;
using Xunit;

namespace TDDMicroExercises.LeaderBoard.Tests
{
    public class LeaderboardTest
    {
        [Fact]
        public void ShouldSumThePoints()
        {
            var results = TestData.SampleLeaderboard1.DriverResults();
            Assert.True(results.ContainsKey("Lewis Hamilton"));
            Assert.Equal(18 + 18 + 25, results["Lewis Hamilton"]);
        }

        [Fact]
        public void ShouldFindTheWinner()
        {
            Assert.Equal("Lewis Hamilton", TestData.SampleLeaderboard1.DriverRankings()[0]);
        }

        [Fact]
        public void ShouldKeepAllDriversWhenSamePoints()
        {
            var winDriver1 = new Race("Australian Grand Prix", TestData.Driver1, TestData.Driver2, TestData.Driver3);
            var winDriver2 = new Race("Malaysian Grand Prix", TestData.Driver2, TestData.Driver1, TestData.Driver3);
            var exEquoLeaderBoard = new Leaderboard(winDriver1, winDriver2);

            var rankings = exEquoLeaderBoard.DriverRankings();

            Assert.Equal(
                new List<string> { TestData.Driver2.Name, TestData.Driver1.Name, TestData.Driver3.Name },
                rankings);
        }
    }
}

File: kotlin/Leaderboard/src/test/kotlin/tddmicroexercises/leaderboard/LeaderboardTest.kt
package tddmicroexercises.leaderboard


import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import tddmicroexercises.leaderboard.TestData.driver1
import tddmicroexercises.leaderboard.TestData.driver2
import tddmicroexercises.leaderboard.TestData.driver3

class LeaderboardTest {

    @Test
    fun itShouldSumThePoints() {
        // setup

        // act
        val results = TestData.sampleLeaderboard1.driverResults()

        // verify
        assertTrue(results.containsKey("Lewis Hamilton"), "results $results")
        assertEquals((18 + 18 + 25).toLong(), (results.get("Lewis Hamilton") as Int).toLong())
    }

    @Test
    fun isShouldFindTheWinner() {
        // setup

        // act
        val result = TestData.sampleLeaderboard1.driverRankings()

        // verify
        assertEquals("Lewis Hamilton", result.get(0))
    }

    @Test
    fun itShouldKeepAllDriversWhenSamePoints() {
        // setup
        // bug, drops drivers with same points
        val winDriver1 = Race("Australian Grand Prix", driver1, driver2, driver3)
        val winDriver2 = Race("Malaysian Grand Prix", driver2, driver1, driver3)
        val exEquoLeaderBoard = Leaderboard(winDriver1, winDriver2)

        // act
        val rankings = exEquoLeaderBoard.driverRankings()

        // verify
        assertEquals(arrayListOf(driver1.name, driver2.name, driver3.name), rankings)
        // note: the order of driver1 and driver2 is JDK/platform dependent
    }

}
